client/state/data-layer/wpcom/theme-filters/index.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * External dependencies
+ */
+import i18n from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import {
+	THEME_FILTERS_REQUEST,
+	THEME_FILTERS_ADD,
+} from 'state/action-types';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { errorNotice } from 'state/notices/actions';
+
+const fetchFilters = ( { dispatch }, action ) => {
+	dispatch( http( {
+		method: 'GET',
+		apiVersion: '1.2',
+		path: '/theme-filters',
+	}, action ) );
+};
+
+const storeFilters = ( { dispatch }, action, next, data ) =>
+	dispatch( { type: THEME_FILTERS_ADD, filters: data } );
```

Comment:
We have a great opportunity here in the data layer to have a real demarcation point between the JSON API and the data as Calypso wants to render it. I'd highly encourage you to think about some kind of mapping function (and maybe validation as well) which can explicitly convert the structural part of the data (but not necessarily the validity of the data itself).

For example, what does the data look like which comes back from this endpoint? What happens if the response comes back as 200 but the response body indicates a failure? We want to make sure that only data which looks like it could be valid makes it past here successfully. (I tried polling this endpoint and got a 404; is it in production yet?)

Below is a very primitive mapping assuming that the API returns an array of filters with a couple properties each

```js
const isValid = ( { filters } ) => isArray( filters );

const fromApi = ( { filters } ) => ( {
	filters: filters.map( ( { filter_name, is_active } ) => ( {
		name: filter_name.replace( 'wpcom/', '' ), 
		isActive: is_active,
	} ) ),
} )
```

Notice that this provides a nicety in that none of the API-specifics (which don't improve our ability to work with and render the data in Calypso) stay at this juncture and never make it into the system. it encourages a more cohesive way to inject data into Calypso whether from an API or from the console dispatcher or from a websocket. Finally, it keeps our reducers laser-focused on the data they are designed to manage: information in Calypso rendered to the users. Reducers are still able to provide their own validation too, by the way, though if we make these guarantees that we'll feed the reducers data in the appropriate shapes then they can focus on things like, "make sure age is positive" and "strip control characters from display name" etc…